docs/reference/builder.md

```diff
@@ -1291,6 +1291,74 @@ The `STOPSIGNAL` instruction sets the system call signal that will be sent to th
 This signal can be a valid unsigned number that matches a position in the kernel's syscall table, for instance 9,
 or a signal name in the format SIGNAME, for instance SIGKILL.
 
+## HEALTHCHECK
+
+The `HEALTHCHECK` instruction has two forms:
+
+* `HEALTHCHECK [OPTIONS] CMD command` (check container health by running a command inside the container)
+* `HEALTHCHECK NONE` (disable any healthcheck inherited from the base image)
+
+The `HEALTHCHECK` instruction tells Docker how to test a container to check that
+it is still working. This can detect cases such as a web server that is stuck in
+an infinite loop and unable to handle new connections, even though the server
+process is still running.
+
+When a container has a healthcheck specified, it has a _health status_ in
+addition to its normal status. This status is initially `starting`. When a
+health check passes, it becomes `healthy`. After a certain number of failures,
+it becomes `unhealthy`.
+
+The options that can appear before `CMD` are:
+
+* `--interval=DURATION` (default: `30s`)
+* `--timeout=DURATION` (default: `30s`)
+* `--grace=DURATION` (default: `30s`)
+* `--retries=N` (default: `1`)
```

Comment:
Because of this retries, does this mean that a container can never go from unhealthy to healthy again?  What happens if there is a network blip and a single check fails but then the network comes back later?  Is my container stuck in unhealthy for the rest of its life until we restart it or its auto killed?  

The whole retry count and exit on unhealthy seems very... intense for a production level feature.  Fail one check and the container dies...

@talex5 Mostly commenting for ppl that helped designed this to weigh in.  ;)